packages/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -0,0 +1,377 @@
+        super('opentelemetry-instrumentation-amqplib', VERSION, Object.assign({}, config));
+    }
+
+    setConfig(config: AmqplibInstrumentationConfig = {}) {
+        this._config = Object.assign({}, config);
+    }
+
+    protected init(): InstrumentationModuleDefinition<typeof amqp> {
+
+        const channelModelModuleFile = new InstrumentationNodeModuleFile<amqp.Channel>(
+            `amqplib/lib/channel_model.js`,
+            ['>=0.5.5'],
+            this.patchChannelModel.bind(this),
+            this.unpatchChannelModel.bind(this)
+        );
+
+        const callbackModelModuleFile = new InstrumentationNodeModuleFile<amqp.Channel>(
+            `amqplib/lib/callback_model.js`,
+            ['>=0.5.5'],
+            this.patchChannelModel.bind(this),
+            this.unpatchChannelModel.bind(this)
+        );
+
+        const connectModuleFile = new InstrumentationNodeModuleFile<amqp.Channel>(
+            `amqplib/lib/connect.js`,
+            ['>=0.5.5'],
+            this.patchConnect.bind(this),
+            this.unpatchConnect.bind(this)
+        );
+
+        const module = new InstrumentationNodeModuleDefinition<typeof amqp>('amqplib', ['>=0.5.5'], undefined, undefined, [
+            channelModelModuleFile,
+            connectModuleFile,
+            callbackModelModuleFile
+        ]);
+        return module;
+    }
+
+    private patchConnect(moduleExports: any) {
+        moduleExports = this.unpatchConnect(moduleExports);
+        this._wrap(moduleExports, 'connect', this._getConnectPatch.bind(this));
+        return moduleExports;
+    }
+
+    private unpatchConnect(moduleExports: any) {
+        if(isWrapped(moduleExports.connect)) {
+            this._unwrap(moduleExports, 'connect');
+        }
+        return moduleExports;
+    }
+
+    private patchChannelModel(moduleExports: any, moduleVersion: string) {
+        this._wrap(moduleExports.Channel.prototype, 'publish', this._getPublishPatch.bind(this, moduleVersion));
+        this._wrap(moduleExports.Channel.prototype, 'consume', this._getConsumePatch.bind(this, moduleVersion));
+        this._wrap(moduleExports.Channel.prototype, 'ack', this._getAckPatch.bind(this, false, EndOperation.Ack));
+        this._wrap(moduleExports.Channel.prototype, 'nack', this._getAckPatch.bind(this, true, EndOperation.Nack));
+        this._wrap(moduleExports.Channel.prototype, 'reject', this._getAckPatch.bind(this, true, EndOperation.Reject));
+        this._wrap(moduleExports.Channel.prototype, 'ackAll', this._getAckAllPatch.bind(this, false, EndOperation.AckAll));
+        this._wrap(moduleExports.Channel.prototype, 'nackAll', this._getAckAllPatch.bind(this, true, EndOperation.NackAll));
+        this._wrap(moduleExports.Channel.prototype, 'emit', this._getChannelEmitPatch.bind(this));
+        return moduleExports;
+    }
+
+    private unpatchChannelModel(moduleExports: any) {
+        if(isWrapped(moduleExports.Channel.prototype.publish)) {
```

Comment:
seems like prettier didn't run (missing space after `if` for example)
make sure to run prettier on everything to keep the code cosistent

Reply:
After investigating the issue, looks like we are running into [an open bug with using husky with lerna monorepos](https://github.com/typicode/husky/issues/677).

Also, it looks like `prettier` did not run for a while in this repo, as running it locally produces many changes in most of our packages.
In addition, `pre-push` hook is running `yarn test`, which to pass, requires few docker runs.

I will open a new issue for that, and PR that is `prettier`ing our current codebase.
If you find a workflow that solves it, let's open a PR.

Reply:
https://github.com/aspecto-io/opentelemetry-ext-js/issues/83